compiler/optimizations/copyPropagation.cpp

```diff
@@ -216,6 +216,34 @@ static bool needsKilling(SymExpr* se, std::set<Symbol*>& liveRefs)
 
     return false;
   }
+  else if (call->isPrimitive(PRIM_VIRTUAL_METHOD_CALL))
+  {
+    // skip the "base" symbol
+    if (se == call->get(1))
+      return false;
+
+    // skip the "_virtual_method_tmp_" argument
+    if (se == call->get(2))
+      return false;
+
+    ArgSymbol* arg = actual_to_formal(se);
+
+    if (arg->intent == INTENT_OUT   ||
+        arg->intent == INTENT_INOUT ||
+        arg->intent == INTENT_REF   ||
+        arg->hasFlag(FLAG_ARG_THIS)) // Todo: replace with arg intent check?
+    {
+      liveRefs.insert(se->symbol());
+      return true;
+    }
+
+    if (isRecordWrappedType(arg->type))
```

Comment:
What's the reason for this check?

If there's a special thing that we need to do w.r.t record-wrapped types, maybe we should add a test for it?

Reply:
Ah, OK. Thanks!